neqo-common/src/qlog.rs

```diff
@@ -144,3 +149,40 @@ pub fn new_trace(role: Role) -> qlog::TraceSeq {
         }),
     }
 }
+
+#[cfg(test)]
+mod test {
+    use const_format::concatcp;
+    use qlog::events::Event;
+    use test_fixture::{new_neqo_qlog, EXPECTED_LOG_HEADER};
+
+    const EV_DATA: qlog::events::EventData =
+        qlog::events::EventData::SpinBitUpdated(qlog::events::connectivity::SpinBitUpdated {
+            state: true,
+        });
+
+    const EXPECTED_LOG_EVENT: &str = concatcp!(
+        "\u{1e}",
+        r#"{"time":0.0,"name":"connectivity:spin_bit_updated","data":{"state":true}}"#,
+        "\n"
+    );
+
+    #[test]
+    fn test_new_neqo_qlog() {
+        let (_log, contents) = new_neqo_qlog();
+        assert_eq!(contents.to_string(), EXPECTED_LOG_HEADER);
+    }
+
+    #[test]
+    fn test_add_event() {
+        let (mut log, contents) = new_neqo_qlog();
+        log.add_event(|| Some(Event::with_time(1.1, EV_DATA)));
+        assert_eq!(
+            contents.to_string(),
+            format!(
+                "{EXPECTED_LOG_HEADER}{}",
+                EXPECTED_LOG_EVENT.replace("\"time\":0.0,", "\"time\":1.1,")
```

Comment:
```suggestion
                "{EXPECTED_LOG_HEADER}{e}",
                e = EXPECTED_LOG_EVENT.replace("\"time\":0.0,", "\"time\":1.1,")
```
With inline items, I prefer to have all of them inline like this.